bigbluebutton-html5/imports/ui/components/whiteboard/service.js

```diff
@@ -38,6 +38,13 @@ async function handleAddedAnnotation({
   Annotations.upsert(query.selector, query.modifier);
 }
 
+const getHasAnnotations = (presentationId) => {
+  const ann = Annotations.find(
+    {},
+  ).fetch();
+  return ann.filter(a => a.whiteboardId.includes(presentationId)).length > 0;
+}
```

Comment:
In a meeting with a lot of annotations this approach will be reeeeeeeally heavy.
Please filter for the `whiteboardId` in the query instead of get all records of this Collections.
And also set `.limit(1)` because we just need to know if it has at least 1 record for this pres.